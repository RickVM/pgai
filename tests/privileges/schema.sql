\set users {bob,fred,alice,jill}

-- check schema privileges
\! rm -f schema.actual
select
  n as "schema"
, u as "user"
, p as "privilege"
, has_schema_privilege(u, n, p) as granted
from unnest(:'users'::text[]) u
cross join unnest(array['ai', 'wiki']) n
cross join unnest(array['create', 'usage']) p
order by n, p, u
\g (format=aligned) schema.actual
