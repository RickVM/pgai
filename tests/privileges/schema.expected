 schema | user  | privilege | granted 
--------+-------+-----------+---------
 ai     | alice | create    | t
 ai     | bob   | create    | f
 ai     | fred  | create    | f
 ai     | jill  | create    | f
 ai     | alice | usage     | t
 ai     | bob   | usage     | f
 ai     | fred  | usage     | f
 ai     | jill  | usage     | f
 wiki   | alice | create    | t
 wiki   | bob   | create    | f
 wiki   | fred  | create    | f
 wiki   | jill  | create    | f
 wiki   | alice | usage     | t
 wiki   | bob   | usage     | f
 wiki   | fred  | usage     | f
 wiki   | jill  | usage     | f
(16 rows)

